internal/command/scale/count_machines.go

```diff
@@ -111,11 +125,24 @@ func runMachinesScaleCount(ctx context.Context, appName string, appConfig *appco
 		switch {
 		case action.Delta > 0:
 			for i := 0; i < action.Delta; i++ {
-				m, err := launchMachine(ctx, action)
+				var volume *api.Volume
+				if i < len(action.Volumes) {
+					if av := action.Volumes[i]; av.Available {
+						av.Available = false
```

Comment:
@dangra so that we don't reuse same volumes across process groups, I went with the idea of mapping `[]api.Volume => []*availableVolume` which has the definition (`availableVolume { *api.Volume, available: bool }`).

So we can check the availability flag and update it after using them so we don't reuse it later with another process group

Reply:
I see it now but didn’t find where the volume allocation consider the "mount's name" when reusing a existent volume. It is like any volume will do as long as it is in the same region.

Reply:
consider the fly.toml from https://community.fly.io/t/mounts-per-process-group-for-apps-v2/12304

it defines different mounts with different volume names per process group

Reply:
Ahhh sure, missed that. I'll get that done